### PR TITLE
Documentation restructuring and cleanup

### LINKED ISSUE #12671: Documentation Restructuring and Cleanup
**Is your enhancement proposal related to a problem? Please describe.**

- Too many entry points in the documentation and split between details and API documentation. 
- Too many intermediate pages making it difficult to find relevant information
- Subsystem documentation is being added adhoc without any structure
- Mix between API documentation, technical overviews and general developer guides

**Describe the solution you'd like**

- Have only one single entry point for API reference
- Put detailed docs and APIs in the same page
- Remove the currently over-populated API documentation and move the API reference to where they belong
- Every API needs to have a page with more details and guides about the API. API docs on their own are not enough.

**Describe alternatives you've considered**

- Using doxygen directly is not feasible.
-

### PR DESCRIPTION
**Pull request by nashif**
Resolves #12671

### CODE REVIEW

**Comment by nashif** [comment]
preview: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/12991/docs/index.html

**Review by dbkinder — COMMENTED** [review]
While I like the doc reorg, I'd like us to have an index.rst in every folder (a web standard).  So instead of reference.rst in the top level folder, I'd like to see an index.rst in the /api folder that's referenced by the top-level index.rst  (as talked about in issue #6776).  You agree?

**Comment by nashif** [comment]
> While I like the doc reorg, I'd like us to have an index.rst in every folder (a web standard). So instead of reference.rst in the top level folder, I'd like to see an index.rst in the /api folder that's referenced by the top-level index.rst (as talked about in issue #6776). You agree?

yeah, thats why it is WIP. Right now I am just showing the structure I want, next step is moving files around and create index pages etc.

File: doc/subsystems/bluetooth/index.rst [hunk 1]
```diff
@@ -72,8 +76,8 @@ Bluetooth stack:
 Qualification Testing
 *********************
 
-PICS Features
-=============
+-PICS Features
+-=============
```
Comment (jhedberg) [inline doc/subsystems/bluetooth/index.rst]: What's going on here? I.e. what are those `-` added to the beginning of the lines?

**Review by jhedberg — DISMISSED** [review]
Bluetooth changes look fine to me, except for the one issue I've commented on separately.

File: doc/subsystems/bluetooth/index.rst (hunk 1 above)
Comment (nashif) [reply doc/subsystems/bluetooth/index.rst]: sharp eyes, no idea how this happened.

**Comment by nashif** [comment]
@dbkinder new structure in place

**Comment by carlescufi** [comment]
@nashif do you have this built somewhere so I can take a  look at the rendered result without having to build myself locally?

**Comment by nashif** [comment]
> @nashif do you have this built somewhere so I can take a look at the rendered result without having to build myself locally?

pointer in 2 comment

**Review by carlescufi — APPROVED** [review]
Just reviewed this side-by-side and I like the new layout

image

**Review by dbkinder — APPROVED** [review]
I like!  I've got a fix for getting the icons on the home page using fontawesome characters, so no licensing issues, once this gets merged.  (We can run a link checker afterward to see if links got missed.)

**Review by jhedberg — APPROVED** [review]

**Comment by jukkar** [comment]
I wonder if it would be possible to change the API formatting a bit. IMHO it is easier to read the API doc if only the function name is highlighted with the color like in this example below. Having both the `Return` and `Parameter` highlighted makes it harder to read the doc. The Return and Parameter could be set in bold instead of color highlighting.

example

**Comment by pfalcon** [comment]
> Having both the `Return` and `Parameter` highlighted makes it harder to read the doc. 

Btw, as we discussed a couple of days ago on Slack with @nashif, it's weird that we have "Return" first, followed by "Parameters". We both admitted that we only now noticed that, so I don't know if it was always like that or some recent settings/them change. If somebody knows how to fix it, would be nice to do. 

> The Return and Parameter could be set in bold instead of color highlighting.

+1

**Comment by nashif** [comment]
> I wonder if it would be possible to change the API formatting a bit.

open an issue please

**Comment by nashif** [comment]
> t's weird that we have "Return" first, followed by "Parameters".

open an issue please, we can look at this later.

**Pull request merged**